src/monitor/pgautofailover--1.3--1.4.sql

```diff
@@ -0,0 +1,782 @@
+ (
+    IN nodeid           int,
+    IN current_state    pgautofailover.replication_state,
+   OUT node_id          int,
+   OUT node_name        text,
+   OUT node_host        text,
+   OUT node_port        int,
+   OUT node_lsn         pg_lsn,
+   OUT node_is_primary  bool
+ )
+RETURNS SETOF record LANGUAGE C STRICT
+AS 'MODULE_PATHNAME', $$get_other_nodes$$;
+
+comment on function pgautofailover.get_other_nodes
+                    (int,pgautofailover.replication_state)
+        is 'get the other nodes in a group, filtering on current_state';
+
+grant execute on function pgautofailover.get_other_nodes
+                          (int,pgautofailover.replication_state)
+   to autoctl_node;
+
+
+DROP FUNCTION pgautofailover.get_coordinator(text);
+
+CREATE FUNCTION pgautofailover.get_coordinator
+ (
+    IN formation_id  text default 'default',
+   OUT node_host     text,
+   OUT node_port     int
+ )
+RETURNS SETOF record LANGUAGE SQL STRICT
+AS $$
+  select nodehost, nodeport
+    from pgautofailover.node
+         join pgautofailover.formation using(formationid)
+   where formationid = formation_id
+     and groupid = 0
+     and goalstate in ('single', 'wait_primary', 'primary')
+     and reportedstate in ('single', 'wait_primary', 'primary');
+$$;
+
+grant execute on function pgautofailover.get_coordinator(text)
+   to autoctl_node;
+
+
+CREATE FUNCTION pgautofailover.get_most_advanced_standby
+ (
+   IN formationid       text default 'default',
+   IN groupid           int default 0,
+   OUT node_id          bigint,
+   OUT node_name        text,
+   OUT node_host        text,
+   OUT node_port        int,
+   OUT node_lsn         pg_lsn,
+   OUT node_is_primary  bool
+ )
+RETURNS SETOF record LANGUAGE SQL STRICT
+AS $$
+   select nodeid, nodename, nodehost, nodeport, reportedlsn, false
+     from pgautofailover.node
+    where formationid = $1
+      and groupid = $2
+      and reportedstate = 'report_lsn'
+ order by reportedlsn desc, health desc
+    limit 1;
+$$;
+
+grant execute on function pgautofailover.get_most_advanced_standby(text,int)
+   to autoctl_node;
+
+DROP FUNCTION IF EXISTS pgautofailover.remove_node(text, int);
+
+CREATE FUNCTION pgautofailover.remove_node
+ (
+   node_id int
+ )
+RETURNS bool LANGUAGE C STRICT SECURITY DEFINER
+AS 'MODULE_PATHNAME', $$remove_node_by_nodeid$$;
+
+comment on function pgautofailover.remove_node(int)
+        is 'remove a node from the monitor';
+
+grant execute on function pgautofailover.remove_node(int)
+   to autoctl_node;
+
+CREATE FUNCTION pgautofailover.remove_node
+ (
+   node_host text,
+   node_port int default 5432
+ )
+RETURNS bool LANGUAGE C STRICT SECURITY DEFINER
+AS 'MODULE_PATHNAME', $$remove_node_by_host$$;
+
+comment on function pgautofailover.remove_node(text,int)
+        is 'remove a node from the monitor';
+
+grant execute on function pgautofailover.remove_node(text,int)
+   to autoctl_node;
+
+CREATE FUNCTION pgautofailover.perform_promotion
```

Comment:
Shouldn this also get a `GRANT EXECUTE` line? This line is also not in the non upgrade file, so it's not really a problem with the migration. However, I don't understand why this one doesn't need it. Same question holds for some of the other UDFs below.

Reply:
This one should yeah. My understanding is that grant are needed when we're accessing the base tables for INSERT/UPDATE/DELETE ; for read-only operations it's fine. Added this grant in both the files.